sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -1043,30 +1044,6 @@ object GpuCast extends Arm {
     }
   }
 
-  def castStringToInts(
-      input: ColumnVector,
-      ansiEnabled: Boolean,
-      dType: DType): ColumnVector = {
-
-    withResource(GpuCast.sanitizeStringToIntegralType(input, ansiEnabled)) { sanitized =>
```

Comment:
GpuCast.sanitizeStringToIntegralType appears to be no longer referenced and should be removed.

Reply:
This seems to be used in the tests https://github.com/NVIDIA/spark-rapids/blob/c79f4a9344156f7a0bb965b4baa06a063f2a6a3c/tests/src/test/scala/com/nvidia/spark/rapids/CastOpSuite.scala#L1007. Does it need to move into the tests or is this no longer needed?

Reply:
That test is about testing `sanitizseStringToIntegralType` which is no longer needed, so the test is no longer needed as well.  It would be good to ensure the special cases that are used for inputs in this test are preserved in the integration tests so we know we're handling these the same way Spark does.

Reply:
I'm too far out of my element to add a scala test for these values. I will have to get some help to get through that. For now, I have removed the code, but my intention is to either personally add a new test or ensure one is added to test these values.

Reply:
Check out the "hand-picked values" tests in CastOpSuite, we can add the appropriate values there if they aren't covered there already.